randovania/games/am2r/layout/hint_configuration.py

```diff
@@ -0,0 +1,32 @@
+import dataclasses
+from enum import Enum
+
+from randovania.bitpacking.bitpacking import BitPackDataclass, BitPackEnum
+from randovania.bitpacking.json_dataclass import JsonDataclass
+from randovania.bitpacking.type_enforcement import DataclassPostInitTypeCheck
+
+
+class ArtifactHintMode(BitPackEnum, Enum):
+    DISABLED = "disabled"
+    HIDE_AREA = "hide-area"
+    PRECISE = "precise"
+
+    @classmethod
+    def default(cls) -> "ArtifactHintMode":
+        return cls.PRECISE
+
+
+class IceBeamHintMode(BitPackEnum, Enum):
+    DISABLED = "disabled"
+    HIDE_AREA = "hide-area"
+    PRECISE = "precise"
+
+    @classmethod
+    def default(cls) -> "IceBeamHintMode":
+        return cls.PRECISE
```

Comment:
why have two identical classes?

Reply:
You mean just using a `ItemHintMode` class, which is the type for both dna and ice beam?
If yes, sound neat, but that should then probably be refactored for Prime1 too, as it was the place where i was copying this all from
https://github.com/randovania/randovania/blob/main/randovania/games/prime1/layout/hint_configuration.py